compiler warning on cc65/scanner.c

As reported on the maillist, this occurs on recent MacOS with clang 15, but also on OpenBSD with clang 16.0.6:

BUILD_ID: Git 5c3ff714a
cc65/scanner.c
cc65/scanner.c:1270:27: warning: cast to smaller integer type 'token_t' (aka 'enum token_t') from 'void *' [-Wvoid-pointer-to-enum-cast]
        token_t LastTok = (token_t)CollLast (C);
                          ^~~~~~~~~~~~~~~~~~~~~
1 warning generated.

--- a/src/cc65/scanner.c
+++ b/src/cc65/scanner.c
@@ -1267,7 +1267,7 @@ static int CloseBrace (Collection* C, token_t Tok)
 */
 {
     if (CollCount (C) > 0) {
-        token_t LastTok = (token_t)CollLast (C);
+        token_t LastTok = (token_t)(intptr_t)CollLast (C);
         if (LastTok == Tok) {
             CollPop (C);
             NextToken ();

